src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -179,7 +179,14 @@ private async Task LoadAll(bool ignoreFailures)
                     {
                         await foreach (ConfigurationSetting setting in _client.GetConfigurationSettingsAsync(selector, CancellationToken.None).ConfigureAwait(false))
                         {
-                            data[setting.Key] = setting;
+                            if (JsonConfigurationParser.IsJsonContentType(setting.ContentType))
+                            {

```

Comment:
This is only one of the multiple places where settings are retrieved from the configuration store. For this change, we can add an adapter that implements `IKeyValueAdapter`, which is designed to run for each key-value, irrespective of where it is retrieved or updated. For examples, you can look at the classes `AzureKeyVaultKeyValueAdapter` and `FeatureManagementKeyValueAdapter`.

Reply:
Thanks for pointing this out! I was looking at KeyVault and FeatureManagement adapters. It looks like they are added to AzureAppConfigurationOptions only when users explicitly configure them with AddAzureAppConfiguration method. In the case of JSON values, we should always add this adapter by default right?

---
In reply to: [453138843](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453138843) [](ancestors = 453138843)

Reply:
`AzureKeyVaultKeyValueAdapter` is added by default, and `FeatureManagementKeyValueAdapter` is added only when user uses `UseFeatureFlags` in their code. For JSON-related adapter, we can enable it by default. It will provide an abstraction over the code to detect if it is JSON and actual processing, so the logic can be at once place, instead of all places where we update settings.

---
In reply to: [453876591](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r453876591) [](ancestors = 453876591,453138843)

Reply:
Done

---
In reply to: [454034139](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454034139) [](ancestors = 454034139,453876591,453138843)